apis/python/src/tiledbsoma/experiment_query/eq_types.py

```diff
@@ -0,0 +1,23 @@
+from typing import Dict, Optional, Sequence
+
+import pyarrow as pa
+from typing_extensions import TypedDict
+
+
+class ExperimentAxisQueryReadArrowResult(TypedDict, total=False):
+    """Return type for the ExperimentAxisQuery.read() method."""
+
+    obs: pa.Table
+    var: pa.Table
+    X: pa.Table
+    X_layers: Dict[str, pa.Table]
+
+
+AxisColumnNames = TypedDict(
+    "AxisColumnNames",
+    {
+        "obs": Optional[Sequence[str]],  # None is all
+        "var": Optional[Sequence[str]],
+    },
+    total=False,
+)
```

Comment:
Any reason for the difference between this TypedDict and the above one? Not a major problem but might be worth making consistent. This would also let you put docstring-style strings below each entry which tooling can pick up on in the future.